### ISSUE TITLE
[Android] How to pass in Android developer payload and oldPurchasedSkus (i.e. extra data)?

### ISSUE THREAD

**Issue by Defcon0** [AlexDisler/cordova-plugin-inapppurchase#215]
Hello,

I have a backend where users can create an account.

Since this is essential in order to avoid restoring subscription purchases for more than one account in my backend service: How can I pass in Android developer payload and oldPurchasedSkus (i.e. extra data) to the Android subscribe interface?

I see in the Android code this is already implemented but I can't see any chance in the JavaScript.

Thanks in advance.

Bye Defcon0

**Comment by lincolnthree** [comment]
I also need to know how to do this. @Defcon0 did you ever figure out a solution? Thanks!

**Comment by Defcon0** [comment]
Unfortunately not. @AlexDisler doesn’t seem to maintain this repo anymore. What a shame!

I ended up using this plugin only for iOS and the other one for Android. Not cool but working.

**Comment by lincolnthree** [comment]
:( Yeah. I'm spending the day trying to figure out https://github.com/j3k0/cordova-plugin-purchase

API is a bit more opaque, but seems like it handles the necessities for recurring subscriptions.

**Comment by lincolnthree** [comment]
@Defcon0 Is that the plugin you ended up using on Android?

**Comment by Defcon0** [comment]
Yup this it. But the plugin threw exceptions over exceptions on iOS for me so I had to use this plugin here for iOS

**Comment by lincolnthree** [comment]
Oh boy :(

**Comment by Defcon0** [comment]
Yup, purchase was by far the most annoying part of the app development. Hopefully we don‘t get some api change and these plugins don’t work anymore
